addons/binding/org.openhab.binding.konnected/src/main/java/org/openhab/binding/konnected/handler/KonnectedHandler.java

```diff
@@ -0,0 +1,330 @@
+            updateState(channelUID, state);
+        }
+
+        // if the pin is an actuator pin process the command
+        if (isAct.get(zone)) {
+
+            Map<String, String> properties = this.thing.getProperties();
+            String host = properties.get("ipAddress");
+            KonnectedHTTPUtils http = new KonnectedHTTPUtils();
+            try {
+                String payload = "{\"pin\":" + pin + ",\"state\":" + scommand + "}";
+                logger.debug("The command payload  is: {}", payload);
+                String data = http.doPut(host + "/device", payload);
+            }
+
+            catch (IOException e) {
+                logger.debug("Getting the state of the pin failed: {}", e);
+            }
+        }
+        // if it is is a sensor pin do nothing
+        else {
+            logger.debug("The command was for a sensor pin so we are ignoring the command");
+        }
+    }
+
+    /**
+     * Process a {@link WebHookEvent that has been received by the servelet from a konnected module
+     *
+     * @param pin the pin number which which was activated
+     *
+     * @param state the state of the pin
+     *
+     * @param setAuth the token sent with the response
+     *
+     */
+
+    public void handleWebHookEvent(String pin, String State, String sentAuth) {
+        // convert pin to zone based on array
+
+        // get the zone number based off of the index location of the pin value
+        String channelid = "Zone_" + Integer.toString(Arrays.asList(PIN_TO_ZONE).indexOf(Integer.parseInt(pin)));
+        logger.debug("The channelid of the event is: {}, the Auth Token is: {}", channelid, sentAuth);
+        StringType channelstate = new StringType(State);
+        if (sentAuth.endsWith(config.Auth_Token)) {
+            updateState(channelid, channelstate);
+        }
+
+        else {
+            logger.debug("The auth token sent did not match what was expected so the state was not accepted.");
+        }
+    }
+
+    @Override
+    public void initialize() {
+        // Long running initialization should be done asynchronously in background.
+        config = getConfigAs(KonnectedConfiguration.class);
+
+        // add the isActuator elements to the boolean array
+        setisAct();
+
+        // Map<String, String> properties = this.thing.getProperties();
+        // String Host = properties.get("ipAddress");
+        logger.debug("Setting up Konnected Module WebHook");
+        webHookServlet.activate(this);
+        updateKonnectedModule();
+
+        updateStatus(ThingStatus.ONLINE);
+
+        // Note: When initialization can NOT be done set the status with more details for further
+        // analysis. See also class ThingStatusDetail for all available status details.
+        // Add a description to give user information to understand why thing does not work
+        // as expected. E.g.
+        // updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
```

Comment:
Can you throw an exception when the initialization fails and then handle it here to update the status to offline?

Reply:
I agree, please initialize the parameters of the thing here once. And if there is an error the thing status should be set to `THING_OFFLINE` with ThingStatusDetail `CONFIGURATION_ERROR`.

At the moment you always set the thing to `ONLINE`...

Reply:
exceptions added now i have several cases added for tracking when the thing is online